src/xpcc/architecture/platform/driver/core/avr/ram.cpp.in

```diff
@@ -76,9 +77,9 @@ initializeMemory(void)
 void *
 xpcc::avr::allocateMemory(std::size_t requestedSize)
 {
-	if (requestedSize == 0 ||
-		requestedSize > MAX_BLOCK_PARTS * BLOCK_SIZE)
+	if (requestedSize == 0 || requestedSize > MAX_BLOCK_PARTS * BLOCK_SIZE)
 	{
+		xpcc_assert(0, "core", "heap", "malloc", requestedSize);
```

Comment:
Hm, ok, good points. I'll turn the C asserts into debug asserts, and add another to C++ `new`.
My reasoning was that very few devs check the malloc return value and so the program would crash, and I wanted to catch that.